synthtool/metadata.py

```diff
@@ -248,19 +263,60 @@ def _get_git_source_map(metadata) -> Dict[str, object]:
     """
     source_map = {}
     for source in metadata.sources:
-        if source.HasField("git"):
+        if source.git:
             git_source = source.git
             source_map[git_source.name] = git_source
     return source_map
 
 
-def _clear_local_paths(metadata):
+def _clear_local_paths():
     """Clear the local_path from the git sources.
 
     There's no reason to preserve it, and it may leak some info we don't
     want to leak in the path.
     """
-    for source in metadata.sources:
-        if source.HasField("git"):
+    for source in _metadata.sources:
+        if source.git:
             git_source = source.git
             git_source.ClearField("local_path")
+
+
+def _add_self_git_source():
+    """Adds current working directory as a git source.
+
+    Returns:
+        The number of git sources added to metadata.
+    """
+    git = shutil.which("git")
+    cwd = os.getcwd()
+    completed_process = subprocess.run([git, "status"])
+    if completed_process.returncode:
+        log.warning("%s is not directory in a git repo.", cwd)
+        return 0
+    completed_process = subprocess.run(
+        [git, "remote", "get-url", "origin"],
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    url = completed_process.stdout.strip()
+    completed_process = subprocess.run(
+        [git, "log", "--no-decorate", "-1", "--pretty=format:%H"],
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    sha = completed_process.stdout.strip()
+    add_git_source(name=".", remote=url, sha=sha, local_path=cwd)
+
+
+def _combine_commit_logs():
+    """
+    Combine the commit logs of all the git sources into one commit log.
+    """
+    text = []
+    for source in _metadata.sources:
+        if source.git and source.git.log:
+            git_source = source.git
+            name = git_source.name
+            text.append(f"Changes in {name}:")
+            text.append(git_source.log)
+    _metadata.combined_commit_log = "\n".join(text)
```

Comment:
I was thinking it might make sense to populate this as an array rather than a `\n` separated string, might make it easier if we want to manipulate the data.